### DISCUSSION TITLE
Possible to use with Openrouter.ai?

### DISCUSSION THREAD

**Discussion by Jirito0** [microsoft/autogen#208] [General]
I'd like to use this with the models supported by Openrouter- the GPT4 32k  in particular to help tackle the token limit for the time being.

Openrouter supports apps that use OAI API keys, and I attempted inserting my Openrouter key, and stated in the OAI Config List to use GPT4-32k, but I just get this error: "OpenAI.error.AuthenticationError: Incorrect API key provided: sk-or-v1-83f***********"

I also reviewed the sections regarding using Local LLMs, but it doesn't provide any information regarding using API's like Openrouter.

**Comment by sonichi** [comment]
I don't have experience with it. I asked on Discord.

**Comment by vanetreg** [comment]
> I'd like to use this with the models supported by Openrouter

Hey @Jirito0 
did it work finally? Could you share details?
I plan to try it now, either directly or via LiteLLM proxy...

**Reply by sonichi** [reply]
@ekzhu @olgavrou @BeibinLi fyi

**Comment by ghost** [comment]
It works for me. 
    {
      "model": "mistralai/mixtral-8x7b-instruct",
      "base_url": "https://openrouter.ai/api/v1",
      "api_key": "sk-or-v1-***********************************"
    },

**Reply by yingdaomayi-qwe** [reply]
Hey, @reevolutiva-it 
where should I add this code?
Thank you!

**Reply by AndreaGelmini** [reply]
I think the answer can be found here. 
https://microsoft.github.io/autogen/stable/user-guide/autogenstudio-user-guide/faq.html

**Reply by thistleknot** [reply]
> It works for me. { "model": "mistralai/mixtral-8x7b-instruct", "base_url": "https://openrouter.ai/api/v1", "api_key": "sk-or-v1-***********************************" },

Error testing model component: Error code: 401 - {'message': 'Unauthorized', 'request_id': '8db2ce0742fb68299392be724edfbb30'}
